src/main/java/minicraft/core/Initializer.java

```diff
@@ -141,6 +146,72 @@ public void windowClosing(WindowEvent e) {
 		});
 
 		frame.setVisible(true);
+		logoSplash.setDisplay(true);
+		logoSplash.renderer.start();
+	}
+
+	private static class LogoSplashCanvas extends JPanel {
+		private Image logo;
+
+		{
+			try {
+				logo = ImageIO.read(Objects.requireNonNull(Initializer.class.getResourceAsStream("/assets/textures/gui/title.png")));
+			} catch (IOException e) {
+				throw new RuntimeException(e);
+			}
+		}
+
+		private int transparency = 255;
+		private boolean display = false;
+		private boolean inAnimation = false;
+		private boolean interruptWhenAnimated = false;
+
+		public Thread renderer = new Thread(() -> {
+			do {
+				repaint();
+				if (interruptWhenAnimated && !inAnimation) break;
+			} while (!Initializer.logoSplash.renderer.isInterrupted());
+		}, "Logo Splash Screen Renderer");
+
+		@Override
+		public void paintComponent(Graphics g) {
+			super.paintComponent(g);
+			if (transparency < 255) g.drawImage(logo, getWidth()/2 - logo.getWidth(frame)*2, getHeight()/2 - logo.getHeight(frame)*2, logo.getWidth(frame)*4, logo.getHeight(frame)*4, frame);
```

Comment:
Having all of this on the same line is confusing. For readability it's better to use curly brackets and move it to the next line.